server/utils.go

```diff
@@ -52,3 +54,13 @@ func getString(key string, props model.StringInterface) string {
 	}
 	return value
 }
+
+func (p *Plugin) getPluginOauthURL() (string, error) {
+	siteURL, err := p.getSiteURL()
```

Comment:
pathEscape the site url as well?

Reply:
@wiggin77 nope because then it will escape the slashes and whatnot.

Example program:
```
package main

import (
	"fmt"
	"net/url"
)


func main() {
	siteUrl := "https://myapp.com"
	pluginId := url.PathEscape("com.mattermost.myplugin")


	fmt.Printf("%s/plugins/%s/oauth2\n", siteUrl, pluginId)

	fmt.Printf("%s/plugins/%s/oauth2\n",url.PathEscape(siteUrl), pluginId)
}
```

Example program's output:
```
https://myapp.com/plugins/com.mattermost.myplugin/oauth2
https:%2F%2Fmyapp.com/plugins/com.mattermost.myplugin/oauth2
```


Which raises a good point that those functions should have tests.